packages/amplify-category-function/src/provider-utils/awscloudformation/utils/functionPluginLoader.ts

```diff
@@ -143,11 +192,23 @@ async function getSelectionFromContributors<T>(context: any, selectionOptions: P
     selection = answer.selection;
   }
 
-  return {
-    value: selection,
-    pluginPath: selectionMap.get(selection).path,
-    pluginId: selectionMap.get(selection).pluginId,
-  };
+  if (Array.isArray(selection)) {
```

Comment:
I would do
```
if (!Array.isArray(selection)) {
  selection = [selection];
}
```
Then run the mapping logic

Reply:
fixed